doc/src/rabbitmq.md

```diff
@@ -34,4 +34,40 @@ After upgrading a cluster, enable all feature flags:
 sudo -u rabbitmq rabbitmqctl enable_feature_flag all
 ```
 
+## Upgrading from NixOS 20.09 while keeping RabbitMQ 3.6.5
+
+To be able to upgrade NixOS 20.09 machines using the `rabbitmq36_5` role,
+we provide a way to keep the unchanged rabbitmq service running after the system
+upgrade. This conserves the specific rabbitmq config for the machine and
+cannot be used on new 23.11 machines.
+
+The upgrade process starts with generating Nix config on the running machine.
+Put the generated config in :file:`/etc/local/nixos/rabbitmq365-frozen.nix`.
+
+```shell
+#!/usr/bin/env sh
+service=$(realpath /etc/systemd/system/rabbitmq.service)
+storePath=${service%%/rabbitmq.service}
+
+cat <<EOF
+# Generated config to freeze the existing rabbitmq unit file and all dependencies
+# to keep it running after an upgrade to 23.11.
+{ lib, ... }:
+{
+  # This no-op option declaration is needed for building the 20.09 system.
+  options.flyingcircus.services.rabbitmq365Frozen.service = lib.mkOption {};
```

Comment:
use role instead of the service

Reply:
The option is actually only declared for the service which makes sense. So keep it as is.